src/algoliasearch.helper.js

```diff
@@ -431,10 +439,184 @@ AlgoliaSearchHelper.prototype.setState = function(newState) {
+    forEach(attributes, function(attr) {
+      if (this.state.isConjunctiveFacet(attr) && this.state.isFacetRefined(attr)) {
+        if (!partialState.facetsRefinements) partialState.facetsRefinements = {};
+        partialState.facetsRefinements[attr] = this.state.facetsRefinements[attr];
+      }
+
+      if (this.state.isDisjunctiveFacet(attr) && this.state.isDisjunctiveFacetRefined(attr)) {
+        if (!partialState.disjunctiveFacetsRefinements) partialState.disjunctiveFacetsRefinements = {};
+        partialState.disjunctiveFacetsRefinements[attr] = this.state.disjunctiveFacetsRefinements[attr];
+      }
+
+      var numericRefinements = this.state.getNumericRefinements(attr);
+      if (!isEmpty(numericRefinements)) {
+        if (!partialState.numericRefinements) partialState.numericRefinements = {};
+        partialState.numericRefinements[attr] = this.state.numericRefinements[attr];
+      }
+    }, this);
+  } else {
+    if (!isEmpty(this.state.numericRefinements)) {
+      partialState.numericRefinements = this.state.numericRefinements;
+    }
+    if (!isEmpty(this.state.facetsRefinements)) partialState.facetsRefinements = this.state.facetsRefinements;
+    if (!isEmpty(this.state.disjunctiveFacetsRefinements)) {
+      partialState.disjunctiveFacetsRefinements = this.state.disjunctiveFacetsRefinements;
+    }
+    if (!isEmpty(this.state.hierarchicalFacetsRefinements)) {
+      partialState.hierarchicalFacetsRefinements = this.state.hierarchicalFacetsRefinements;
+    }
+  }
+
+  var searchParameters = filter(
+    filters,
+    function(f) { return f.indexOf('attribute:') === -1; });
+
+  forEach(
+    searchParameters,
+    function(parameterKey) {
+      partialState[parameterKey] = this.state[parameterKey];
+    },
+    this
+  );
+
+  return partialState;
+};
+
+var shortener = require('./SearchParameters/shortener');
+
+/**
+ * Get part of the state as a query string. By default, the output keys will not
+ * be prefixed and will only take the applied refinements and the query.
+ * @param {object} [options] May contain the following parameters :
+ *  - filters : possible values are all the keys of the {SearchParameters}, 'index' for the index,
+ *    all the refinements with 'attribute:*' or for some specific attributes with 'attribute:theAttribute'
+ *  - prefix : prefix in front of the keys
+ *  - moreAttributes : more values to be added in the query string. Those values
+ *    won't be prefixed.
+ * @return {string} the query string
+ */
+AlgoliaSearchHelper.prototype.getStateAsQueryString = function getStateAsQueryString(options) {
+  var filters = options && options.filters || ['query', 'attribute:*'];
+  var moreAttributes = options && options.moreAttributes;
+  var prefixForParameters = options && options.prefix || '';
+
+  var partialState = this.getState(filters);
+
+  var encodedState = mapKeys(
+    partialState,
+    function(v, k) {
+      var shortK = shortener.encode(k);
```

Comment:
We have a test to check the exhaustivity of the shorten versions of the keys, so we should have a shorten version for each key in the searchparameters.